apps/client/src/features/viewers/lower-thirds/LowerThird.tsx

```diff
@@ -0,0 +1,246 @@
+import { useEffect, useMemo, useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import { Message, OntimeEvent, ViewSettings } from 'ontime-types';
+
+import { overrideStylesURL } from '../../../common/api/apiConstants';
+import NavigationMenu from '../../../common/components/navigation-menu/NavigationMenu';
+import { LOWER_THIRD_OPTIONS } from '../../../common/components/view-params-editor/constants';
+import ViewParamsEditor from '../../../common/components/view-params-editor/ViewParamsEditor';
+import { useRuntimeStylesheet } from '../../../common/hooks/useRuntimeStylesheet';
+import { cx } from '../../../common/utils/styleUtils';
+
+import './LowerThird.scss';
+
+enum srcKeys {
+  Title = 'title',
+  Subtitle = 'subtitle',
+  Presenter = 'presenter',
+  LowerMsg = 'lowerMsg',
+}
+
+enum triggerType {
+  Event = 'event',
+  Manual = 'manual',
+}
+
+type LowerOptions = {
+  width: number;
+  trigger: triggerType;
+  topSrc: srcKeys;
+  bottomSrc: srcKeys;
+  topColour: string;
+  bottomColour: string;
+  topBg: string;
+  bottomBg: string;
+  topSize: string;
+  bottomSize: string;
+  transition: number;
+  delay: number;
+  key: string;
+  lineColour: string;
+  lineHeight: string;
+};
+
+interface LowerProps {
+  eventNow: OntimeEvent | null;
+  viewSettings: ViewSettings;
+  lower: Message;
+}
+
+const defaultOptions: Readonly<LowerOptions> = {
+  width: 45,
+  trigger: triggerType.Event,
+  topSrc: srcKeys.Title,
+  bottomSrc: srcKeys.Subtitle,
+  topColour: '000000ff',
+  bottomColour: '000000ff',
+  topBg: '00000000',
+  bottomBg: '00000000',
+  topSize: '65px',
+  bottomSize: '40px',
+  transition: 3,
+  delay: 3,
+  key: 'ffffffff',
+  lineColour: 'ff0000ff',
+  lineHeight: '0.4em',
+};
+
+export default function LowerThird(props: LowerProps) {
+  const { eventNow, lower, viewSettings } = props;
+  const [searchParams] = useSearchParams();
+
+  const options = useMemo(() => {
+    const newOptions = { ...defaultOptions };
+
+    const width = searchParams.get('width');
+    if (width !== null) {
+      newOptions.width = Number(width);
+    }
+
+    const trigger = Object.values(triggerType).find((s) => s === searchParams.get('trigger'));
+    if (trigger) {
+      newOptions.trigger = trigger;
+    }
+
+    const topSrc = Object.values(srcKeys).find((s) => s === searchParams.get('top-src'));
+    if (topSrc) {
+      newOptions.topSrc = topSrc;
+    }
+
+    const bottomSrc = Object.values(srcKeys).find((s) => s === searchParams.get('bottom-src'));
+    if (bottomSrc) {
+      newOptions.bottomSrc = bottomSrc;
+    }
```

Comment:
as a small optimization, you could run the loop only once (with a forEach) maybe?

Because like this it ends up looping through the array three times, when one would probably be enough.

This is small though since I'm assuming `Object.values(srcKeys)` is a small array, but would be more problematic with larger arrays

Reply:
There are only 4 keys, but I will keep it in mide for the future